packages/dds/map/src/directory.ts

```diff
@@ -2447,10 +2447,11 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 	}
 
 	private undeleteSubDirectoryTree(directory: SubDirectory): void {
-		// Restore deleted subdirectory tree. This will unmark "deleted" from the subdirectories from bottom to top.
-		for (const [_, subDirectory] of this._subdirectories.entries()) {
-			this.undeleteSubDirectoryTree(subDirectory);
-		}
+		// Restore deleted subdirectory tree. Need to undispose the current directory first, then get access to the iterator.
+		// This will unmark "deleted" from the subdirectories from top to bottom.
```

Comment:
Why does the order need to change to top to bottom? That seems unrelated to the infinite calls.

Reply:
@scarlettjlee The order does not matter, but it is necessary to undispose the current level directory first before accessing the iterator, consequently the undispose order becomes top-to-bottom